InstantSearch.js/routing-seo-friendly/src/search-router.js

```diff
@@ -0,0 +1,101 @@
+/* global instantsearch */
+
+function getCategorySlug(name) {
+  return name.replace(/\s/g, '+');
+}
+
+function getCategoryName(slug) {
+  return slug.replace(/\+/g, ' ');
+}
+
+const router = instantsearch.routers.history({
+  windowTitle({ category, query }) {
+    const queryTitle = query ? `Results for "${query}"` : 'Search';
+
+    if (category) {
+      return `${category} – ${queryTitle}`;
+    }
+
+    return queryTitle;
+  },
+
+  createURL({ qsModule, routeState, location }) {
+    let baseUrl = location.href.split('/search/')[0];
+
+    if (baseUrl[baseUrl.length - 1] !== '/') {
+      baseUrl += '/';
+    }
+
+    const categoryPath = routeState.category
+      ? `${getCategorySlug(routeState.category)}/`
+      : '';
+    const queryParameters = {};
+
+    if (routeState.query) {
+      queryParameters.query = encodeURIComponent(routeState.query);
+    }
+    if (routeState.page !== 1) {
+      queryParameters.page = routeState.page;
+    }
+    if (routeState.brands) {
+      queryParameters.brands = routeState.brands.map(encodeURIComponent);
+    }
+
+    const queryString = qsModule.stringify(queryParameters, {
+      addQueryPrefix: true,
+      arrayFormat: 'repeat',
+    });
+
+    return `${baseUrl}search/${categoryPath}${queryString}`;
+  },
+
+  parseURL({ qsModule, location }) {
+    const pathnameMatches = location.pathname.match(/search\/(.*?)\/?$/);
+    const category = getCategoryName(
+      (pathnameMatches && pathnameMatches[1]) || ''
+    );
+    const { query = '', page, brands = [] } = qsModule.parse(
```

Comment:
how come you don't pass the same options as to the stringify?

Reply:
What do you mean?

Reply:
you don't pass arrayMerge here, is it taken automatically?

Reply:
Where do you see such an option in `qs`?

That's how it's transformed back into an array if needed:

https://github.com/algolia/doc-code-samples/blob/0d1464f5b9af7997ebf9975e5c702044c916d1a9/InstantSearch.js/routing-seo-friendly/src/search-router.js#L57-L61